src/app/elements/element-selection-tab/select-element/select-element-dialog/select-element-dialog.component.ts

```diff
@@ -11,19 +14,29 @@ import {ElementsComponent} from "../../../elements.component";
 })
 @Injectable()
 export class SelectElementDialogComponent implements OnInit{
+
   public label: string;
   public selectOptionValues: string[] = [];
+  public bufferArray: string[] = [];
   private currentIndex:number;
 
-  constructor(public dialogRef: MatDialogRef<SelectElementDialogComponent>, @Inject(MAT_DIALOG_DATA) public data: DialogData, private dataService: Service) {
+  constructor(@Inject(MAT_DIALOG_DATA) public data: DialogData) {
     this.label = data.label;
     this.selectOptionValues = data.selectOptionValues;
     this.currentIndex = 0;
   }
 
   ngOnInit() {
+    for (let i = 0; i < this.selectOptionValues.length; i++) {
+      this.bufferArray.push(this.selectOptionValues[i])
+    }
```

Comment:
Any reason this is done in `ngOnInit` instead of inside the constructor? Also, because the array contains a primitive type you could do
```typescript
this.bufferArray = [...this.selectOptionValues]
```

Reply:
> this.bufferArray = [...this.selectOptionValues]

⬆️ I've just learned something new, thank you! 
No specific reason for the `ngOnInit`, I just made a mistake. Relocated the functionality to the constructor. Thank you